test/fast-list-test.js

```diff
@@ -349,4 +407,65 @@ suite('FastList >', function() {
       item.dispatchEvent(clickEvt);
     });
   });
+
+  suite('When content is late', function() {
+    var fastList, populatePromise;
+
+    setup(function() {
+      populatePromise = new MockPromise();
+      sinon.stub(source, 'populateItem').returns(populatePromise.promise);
+      sinon.stub(source, 'populateItemDetail').returns(false);
+      fastList = new FastList(source);
+      scheduler.mutation.yield();
+    });
+
+    test('it should populate fully once the promise resolves', function() {
+      var items = container.querySelectorAll('ul li');
+      for (var i = 0; i < items.length; i++) {
+        var item = items[i];
+        assert.equal(item.dataset.populated, 'false');
+      }
+
+      source.populateItem.restore();
+      source.populateItemDetail.restore();
+      populatePromise.resolve();
+
+      return Promise.resolve().then(function() {
+        assertCurrentlyRenderedWindow({
+          container: container,
+          source: source,
+          from: 0,
+          to: 21,
+          expectsDetail: true
+        });
+      });
+    });
+
+    test('it should reveal the item with a transition', function() {
+      var trPromise = new MockPromise();
+      var items = container.querySelectorAll('ul li');
+      sinon.stub(scheduler, 'transition').returns(trPromise.promise);
+
+      source.populateItem.restore();
+      source.populateItemDetail.restore();
+      populatePromise.resolve();
+
+      return Promise.resolve().then(function() {
+        scheduler.transition.yield();
+
+        for (var i = 0; i < items.length; i++) {
+          var item = items[i];
+          assert.equal(item.dataset.populated, 'true');
+        }
+
+        trPromise.resolve();
+        return Promise.resolve().then(function() {
+          for (var i = 0; i < items.length; i++) {
+            var item = items[i];
+            assert.equal(item.style.transition, '');
+          }
+        });
+      });
+    });
```

Comment:
yeah this part is "scheduled for plugin extraction", but it feels safer having the tests first :)